### PR TITLE
Profiles on login rendering 

### PR DESCRIPTION
**Pull request by harrietmorris**

### CODE REVIEW

File: client/services/apiService.ts [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
-const BASE_URL = 'http://localhost:3000'; //this may need to change to IP address
+//const BASE_URL = 'http://localhost:3000'; //this may need to change to IP address
+const BASE_URL = 'http://10.0.2.2:3000'; //this is the URL used for android simulator
 
 export const getUser = async (id: number): Promise<UserType> => {
   try {
     const response: AxiosResponse<UserType> = await axios.get(`${BASE_URL}/users/${id}`);
-    console.log('getting user:', response.data);
+    
+    // console.log('getting user:', response.data);
```
Comment (sebastiandve) [inline client/services/apiService.ts:11]: remove this comment

File: client/services/apiService.ts [hunk 2]
```diff
@@ -1,20 +1,31 @@
...
 };
 
-// export const getAllProfiles = async (userId: number): Promise<ProfileType[]> => {};
+export const getAllProfiles = async (userId: number): Promise<ProfileType[]> => {
+  try {
+    const response: AxiosResponse<ProfileType[]> = await axios.get(`${BASE_URL}/users/${userId}/profiles`);
+    // console.log('getting all profiles:', response.data)
+    return response.data;
+} catch (e) {
+    console.log('error getting profiles', e);
```
Comment (sebastiandve) [inline client/services/apiService.ts:25]: this should be `console.error`

File: client/services/apiService.ts [hunk 3]
```diff
@@ -1,20 +1,31 @@
...
     console.log('error getting user', e);
     throw e;
   }
 };
 
-// export const getAllProfiles = async (userId: number): Promise<ProfileType[]> => {};
+export const getAllProfiles = async (userId: number): Promise<ProfileType[]> => {
+  try {
+    const response: AxiosResponse<ProfileType[]> = await axios.get(`${BASE_URL}/users/${userId}/profiles`);
+    // console.log('getting all profiles:', response.data)
```
Comment (sebastiandve) [inline client/services/apiService.ts:22]: remove comment

File: client/services/apiService.ts [hunk 4]
```diff
@@ -1,20 +1,31 @@
...
+    return response.data;
+} catch (e) {
+    console.log('error getting profiles', e);
+    throw e;
```
Comment (sebastiandve) [inline client/services/apiService.ts:26]: we should not throw an error, thats the point of using try/catch

File: client/services/apiService.ts [hunk 5]
```diff
@@ -1,20 +1,31 @@
...
 export const getUser = async (id: number): Promise<UserType> => {
   try {
     const response: AxiosResponse<UserType> = await axios.get(`${BASE_URL}/users/${id}`);
-    console.log('getting user:', response.data);
+    
+    // console.log('getting user:', response.data);
     return response.data;
   } catch (e) {
     console.log('error getting user', e);
     throw e;
```
Comment (sebastiandve) [inline client/services/apiService.ts:15]: same thing here

File: client/components/Profiles/ListedProfiles.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        setSelectedProfile(profile);
+        console.log('profile selected:', profile)
+      };
+
+    if (!user) {
+        return <Text>Loading user data...</Text>
+    }
+
+    if (!profiles || profiles.length === 0) {
+        return <Text>No profiles available, please create new profile.</Text>
```
Comment (sebastiandve) [inline client/components/Profiles/ListedProfiles.tsx:35]: TODO: should redirect to create profile page

File: client/components/Profiles/ListedProfiles.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
+import { View, Text, FlatList, StyleSheet } from 'react-native'
+import React, { useEffect } from 'react'
+import { useDataContext } from '@/context/globalContext';
+import { getAllProfiles } from '@/services/apiService';
+import ProfileButton from './ProfileButton';
+import { ProfileType } from '@/types';
+
+
+const ListedProfiles = () => {
+    const { user, profiles, setProfiles, setSelectedProfile } = useDataContext();;
```
Comment (sebastiandve) [inline client/components/Profiles/ListedProfiles.tsx:10]: `;;` just need one

File: client/app/newProfileScreen.tsx [hunk 8]
```diff
@@ -5,6 +5,7 @@ const newProfile = () => {
   return (
     <View>
       <Text>newProfile</Text>
+
```
Comment (sebastiandve) [inline client/app/newProfileScreen.tsx:8]: remove this

File: client/app/profilesScreen.tsx [hunk 9]
```diff
@@ -1,14 +1,61 @@
...
+    <SafeAreaView style={styles.container}>
+      < NewProfileButton 
+        route="/newProfileScreen"
+        buttonStyle={styles.buttonStyle}
+        textStyle={styles.textStyle} />
+      
+      <ListedProfiles />
+
+
+    </SafeAreaView>
```
Comment (sebastiandve) [inline client/app/profilesScreen.tsx:37]: no need for blank lines

**Review by sebastiandve — CHANGES_REQUESTED** [review]
caught some small issue, looks good otherwise

File: client/services/apiService.ts (hunk 4 above)
Comment (harrietmorris) [reply client/services/apiService.ts:26]: It needs to be a throw error or else the Promise<UserType> starts screaming

File: client/components/Profiles/ListedProfiles.tsx (hunk 6 above)
Comment (harrietmorris) [reply client/components/Profiles/ListedProfiles.tsx:35]: If you refer to the profilesScreen.tsx, you'll see the create new profile button is rendered directly above this text

**Pull request closed without merging**